### DISCUSSION TITLE
Job WithoutOverlapping Middleware not working

### DISCUSSION THREAD

**Discussion by anabeto93** [laravel/framework#41974] [Q&A]
- Laravel Version: 9.8.1
- PHP Version: 8.1.2
- Database Driver & Version: sqlite

### Description:
The `middleware()` method on jobs not getting called. Have tried to `dd()` and it does not work unlike `uniqueId()` method when implementing the `ShouldBeUnique` interface.

### Steps To Reproduce:
On a clean laravel app, 
1. Create a job, that simply logs any string
2. Inlcude the `middleware()` method and include the `WithoutOverlapping` middleware, defining a key for it.
3. Test it using phpunit or postman making sure to `dd()` in the middleware

Below are some screenshots to better explain
1. Route endpoint
![route_endpoint](https://user-images.githubusercontent.com/7607657/163293859-78831412-ba6f-46c4-be1f-1c2728aef750.png)
```php
Route::get('log', function() {
    DB::beginTransaction();

    SimpleLogJob::dispatch()->afterCommit()->afterResponse();

    User::create([
        'email' => 'test@mailinator.com',
        'password' => 'unhashedpassword',
        'name' => 'Testing Commit',
    ]);

    DB::commit();

    return response()->json(['message' => "Logging works."]);
});
```
![job_function](https://user-images.githubusercontent.com/7607657/163294012-593a5a9b-cc5a-4a33-8278-2747bac9645e.png)

2. Tests
![test_endpoint](https://user-images.githubusercontent.com/7607657/163294153-e2923b98-4263-4573-9512-f01aef8cb272.png)

```php
public function test_job_dispatched_after_response()
    {
        $response = $this->json('GET', 'api/log');

        $response->assertStatus(200)->assertJson([
            'message' => 'Logging works.'
        ]);

        $users = User::get();

        $this->assertCount(1, $users);
    }
```

3. Result
![test_result](https://user-images.githubusercontent.com/7607657/163294267-6f866494-2187-474f-9bce-04e45d3e351b.png)

Should've died and dumped the result `some_middleware` when the tests run.

**Comment by driesvints** [comment]
@anabeto93 I moved your question here because I think this is a support question, not a bug. Please try some support channels first.

**Comment by datpmwork** [comment]
@anabeto93 

I ran into this question today as well. But after checking the framework's code, I realized.

When dispatching the Job, the `middleware` method will not be triggered by any place.

This method would be called when the job is handled, which is `CallQueueHandler`

https://github.com/laravel/framework/blob/8.x/src/Illuminate/Queue/CallQueuedHandler.php#L114-L123
